### PR TITLE
perf(formatter): use cow and put cow_ascii on clippy

### LINKED ISSUE #5639: perf(formatter): can `cow_to_ascii_lowercase` be used here?
[https://github.com/oxc-project/oxc/blob/main/crates/oxc_prettier/src/format/mod.rs#L1406-L1409](https://github.com/oxc-project/oxc/blob/main/crates/oxc_prettier/src/format/mod.rs#L1406-L1409)

related examples: #5637 

I tried it and it seems to work. Should this be done?

### PR DESCRIPTION
**Pull request by heygsc**
close: #5639 

part of: #5586

### CODE REVIEW

File: crates/oxc_prettier/src/format/mod.rs [hunk 1]
```diff
@@ -1341,7 +1342,7 @@ impl<'a> Format<'a> for NumericLiteral<'a> {
             let mut string = Cow::Borrowed(raw);
 
             if string.contains(|c: char| c.is_ascii_uppercase()) {
```
Comment (Boshen) [inline crates/oxc_prettier/src/format/mod.rs:1344]: Why was this `is_ascii_uppercase` but the code is `cow_to_ascii_lowercase` 🤣 

But these `contains` check can be removed.

Comment (heygsc) [reply crates/oxc_prettier/src/format/mod.rs:1344]: remove all or remove this?

File: crates/oxc_prettier/src/format/mod.rs [hunk 2]
```diff
@@ -1405,7 +1402,7 @@ impl<'a> Format<'a> for BigIntLiteral<'a> {
         let text = self.span.source_text(p.source_text);
         // Perf: avoid a memory allocation from `to_ascii_lowercase`.
         if text.contains(|c: char| c.is_lowercase()) {
```
Comment (Boshen) [inline crates/oxc_prettier/src/format/mod.rs:1404]: Same here?

File: .clippy.toml [hunk 3]
```diff
@@ -1 +1,6 @@
 ignore-interior-mutability = ["oxc_linter::rule::RuleWithSeverity"]
+
+disallowed-methods = [
+  { path = "str::to_ascii_lowercase", reason = "avoid a memory allocation from `to_ascii_lowercase`" },
```
Comment (Boshen) [inline .clippy.toml:4]: Avoid memory allocation. Use `cow_utils::CowUtils::cow_to_ascii_uppercase` instead.

File: crates/oxc_prettier/src/format/mod.rs (hunk 2 above)
Comment (heygsc) [reply crates/oxc_prettier/src/format/mod.rs:1404]: Then the formatter doesn't need cow at the moment🤣

Comment (heygsc) [reply crates/oxc_prettier/src/format/mod.rs:1404]: `if contain` condition is deleted, will the contents still be retained? 

This `fn format` seems there's nothing left.

File: crates/oxc_prettier/src/format/mod.rs [hunk 4]
```diff
@@ -1340,10 +1340,6 @@ impl<'a> Format<'a> for NumericLiteral<'a> {
             let raw = self.span.source_text(p.source_text);
             let mut string = Cow::Borrowed(raw);
 
-            if string.contains(|c: char| c.is_ascii_uppercase()) {
-                string = Cow::Owned(string.to_ascii_lowercase());
```
Comment (Boshen) [inline crates/oxc_prettier/src/format/mod.rs:1344]: oh ... this should be changed to `string = string.to_ascii_lowercase()`?

Comment (heygsc) [reply crates/oxc_prettier/src/format/mod.rs:1344]: Yes. I misunderstood "remove contains check".

**Pull request merged**